### ISSUE TITLE
Progress bar overflows when using gradient accumulation with max_steps

### ISSUE THREAD

**Issue by Enucatl** [fishaudio/fish-speech#1226]
### Self Checks

- [x] This template is only for bug reports. For questions, please visit [Discussions](https://github.com/fishaudio/fish-speech/discussions).
- [x] I have thoroughly reviewed the project documentation (installation, training, inference) but couldn't find information to solve my problem. [English](https://speech.fish.audio/) [中文](https://speech.fish.audio/zh/) [日本語](https://speech.fish.audio/ja/) [Portuguese (Brazil)](https://speech.fish.audio/pt/)
- [x] I have searched for existing issues, including closed ones. [Search issues](https://github.com/fishaudio/fish-speech/issues)
- [x] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/fishaudio/fish-speech/issues/515)).
- [x] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [x] Please do not modify this template and fill in all required fields.

### Cloud or Self Hosted

Self Hosted (Source)

### Environment Details

fine tuning configuration with `max_steps` and `accumulate_grad_batches > 1`

## Description

   When training with `accumulate_grad_batches > 1` and `max_steps` (i.e. `max_epochs=-1`), the default TQDM progress bar overflows past 100% and loses its total, making it impossible to track training progress.

   ## Steps to reproduce

   Train a LoRA with gradient accumulation and a step limit:

   ```bash
   python fish_speech/train.py --config-name text2semantic_finetune \
       project=test \
       +lora@model.model.lora_config=r_8_alpha_16 \
       trainer.max_steps=1 \
       trainer.accumulate_grad_batches=4
   ```

### ✔️ Expected Behavior

   Bar reaches 100% and stops:
   ```
   Epoch 0: 100%|██████████| 4/4
   ```
   (4 forward passes for 1 optimizer step with accumulate=4)

### ❌ Actual Behavior

   Bar overflows past its total and the total disappears:
   ```
   Epoch 0: 100%|██████████| 1/1  ← hits 100% after 1 forward pass
   Epoch 0:                | 2/?  ← total is lost, bar keeps going
   Epoch 0:                | 3/?
   Epoch 0:                | 4/?  ← stops here (4× the displayed total)
   ```
